File: erlang/pangram/src/pangram.erl
-module(pangram).

-export([is_pangram/1]).


is_pangram(_Sentence) -> _s2 = prep(_Sentence), _count=unique_count(_s2), _count == 26.

% Prep the string by lowercasing it and filtering out anything except english letters
prep(_Sentence) -> _S2 = string:to_lower(_Sentence), lists:filter(fun(X) -> is_alpha(X) end, _S2).

is_alpha(_Char) when _Char >= $a, _Char =< $z -> true;
is_alpha(_) -> false.

unique_count(_Sentence) -> sets:size(sets:from_list(_Sentence)).

File: csharp/pangram/Pangram.cs
using System;
using System.Collections.Generic;
using System.Linq;

public static class Pangram
{
    public static bool IsPangram(string input)
    {
        // Pangram = The size of the set of lowercase letters in the input is 26
        return new HashSet<char>(input.ToLower().Where(c => char.IsLetter(c))).Count == 26;
    }
}
